pandas/core/indexes/interval.py

```diff
@@ -556,8 +556,12 @@ def is_non_overlapping_monotonic(self):
         # must be increasing  (e.g., [0, 1), [1, 2), [2, 3), ... )
         # or decreasing (e.g., [-1, 0), [-2, -1), [-3, -2), ...)
         # we already require left <= right
-        return ((self.right[:-1] <= self.left[1:]).all() or
-                (self.left[:-1] >= self.right[1:]).all())
+        if self.closed == 'both':
+            return ((self.right[:-1] < self.left[1:]).all() or
+                    (self.left[:-1] > self.right[1:]).all())
+        else:
```

Comment:
Working on this now. Why would the neither case short circuit to False? Can't we have intervals [(0, 1), (1, 2)] where it is non-overlapping and also cases where it is overlapping [(0, 1), (.5, 3)]

Thanks!